### PR TITLE
Number of fields in Instrument table

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds the number of fields to the Instrument table.

### CODE REVIEW

File: skyportal/models/instrument.py [hunk 1]
```diff
@@ -214,6 +372,16 @@ def api_class_obsplan(self):
...
+    @property
+    def number_of_fields(self):
+        if not self.has_fields:
+            return 0
+        stmt = sa.select(InstrumentField).where(
+            InstrumentField.instrument_id == self.id
+        )
+        count_stmt = sa.select(func.count()).select_from(stmt.distinct())
+        return DBSession().execute(count_stmt).scalar()
+
```
Comment (Theodlz) [inline skyportal/models/instrument.py:384]: Do you need a query here? Given that the instrument has a relationship on fields, I'm wondering if you can't just check the `len(self.fields)` or something similar instead. Worth trying.

**Review by Theodlz — COMMENTED** [review]
Looks good. Just one comment

File: skyportal/models/instrument.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/models/instrument.py:384]: self.fields will load all that data, which I think we want to avoid.

Comment (Theodlz) [reply skyportal/models/instrument.py:384]: Ok that makes sense. LGTM

**Review by Theodlz — APPROVED** [review]
LGTM

**Pull request merged**